flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/StreamTaskNetworkInputTest.java

```diff
@@ -120,67 +112,10 @@ public void testNoDataProcessedAfterCheckpointBarrier() throws Exception {
 		assertEquals(0, output.getNumberOfEmittedRecords());
 	}
 
-	@Test
-	public void testSnapshotWithTwoInputGates() throws Exception {
```

Comment:
why was this test removed?

Reply:
I added this test for FLINK-18139 - input gate index issues in `Unaligner#hasInflightData`, which is called by `StreamTaskNetworkInput`, hence the test here.

However, in this commit, `hasInflightData` is removed. Spilling is fully encapsulated in `InputChannel` + `CheckpointBarrierUnaligner`/`CheckpointedInputGate` and this interaction is already covered in tests at `CheckpointBarrierUnalignerTest`.